src/GraphQL.ApiTests/ApiApprovalTests.PublicApi.GraphQL.Extensions.DI.Microsoft.approved.txt

```diff
@@ -0,0 +1,72 @@
+        public GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Resolve(System.Func<GraphQL.IResolveFieldContext<TSourceType>, T1, T2, T3, T4, TReturnType> resolver) { }
+        public GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> ResolveAsync(System.Func<GraphQL.IResolveFieldContext<TSourceType>, T1, T2, T3, T4, System.Threading.Tasks.Task<TReturnType>> resolver) { }
+        public GraphQL.Extensions.DI.Microsoft.ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4> WithScope() { }
+        public GraphQL.Extensions.DI.Microsoft.ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5> WithType<T5>() { }
+    }
+    public class ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5>
+    {
+        public ResolverBuilder(GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> builder, bool scoped) { }
+        public GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Resolve(System.Func<GraphQL.IResolveFieldContext<TSourceType>, T1, T2, T3, T4, T5, TReturnType> resolver) { }
+        public GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> ResolveAsync(System.Func<GraphQL.IResolveFieldContext<TSourceType>, T1, T2, T3, T4, T5, System.Threading.Tasks.Task<TReturnType>> resolver) { }
+        public GraphQL.Extensions.DI.Microsoft.ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5> WithScope() { }
+    }
+    public class ScopedAsyncFieldResolver<TReturnType> : GraphQL.Resolvers.AsyncFieldResolver<TReturnType>
+    {
+        public ScopedAsyncFieldResolver(System.Func<GraphQL.IResolveFieldContext, System.Threading.Tasks.Task<TReturnType>> resolver) { }
+    }
+    public class ScopedAsyncFieldResolver<TSourceType, TReturnType> : GraphQL.Resolvers.AsyncFieldResolver<TSourceType, TReturnType>
+    {
+        public ScopedAsyncFieldResolver(System.Func<GraphQL.IResolveFieldContext<TSourceType>, System.Threading.Tasks.Task<TReturnType>> resolver) { }
+    }
+    public static class ScopedFieldBuilderExtensions
+    {
+        public static GraphQL.Extensions.DI.Microsoft.ResolverBuilder<TSourceType, TReturnType> Resolve<TSourceType, TReturnType>(this GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> builder) { }
+        public static GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> ResolveScoped<TSourceType, TReturnType>(this GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> builder, System.Func<GraphQL.IResolveFieldContext<TSourceType>, TReturnType> resolver) { }
+        public static GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> ResolveScopedAsync<TSourceType, TReturnType>(this GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> builder, System.Func<GraphQL.IResolveFieldContext<TSourceType>, System.Threading.Tasks.Task<TReturnType>> resolver) { }
+    }
+    public class ScopedFieldResolver<TReturnType> : GraphQL.Resolvers.FuncFieldResolver<TReturnType>
```

Comment:
Do all these resolver subsclasses need to be public?

Reply:
By the way you forgot to delete this file.

Reply:
Oops

Reply:
All of the fieldbuilder classes need to be public, yes, unless we make interfaces for them all, but I don't see any point to that at this time.

Reply:
I deleted the extra file.

Reply:
I talked about resolvers subclasses, notabout builders.

Reply:
No.  I want them public.

Reply:
In #1571  already linked you to a sample in my code where I would use them directly.

Reply:
Yes, I remembered.